### DISCUSSION TITLE
System.DllNotFoundException: Unable to load shared library 'Magick.Native-Q8-x64.dll' or one of its dependencies.

### DISCUSSION THREAD

**Discussion by Manish082007** [dlemstra/Magick.NET#1303] [Help] [answered]
I am not able to resize images using Magick.Net on Debian 11. It is working fine on windows. I am using Amazon ECS Services and deploying a docker image on ECS Service. I am using `Magick.NET-Q8-AnyCPU` library file version `12.2.1`.

**_Let me know if i am missing anything or doing something wrong.
Your help is highly appreciated. I tried many other libraries too but no success.
I have not been able to resize simple images using Asp.Net core application on Linux for the last 3 days.
THANK YOU..._**

### Already doing
1. I am already adding below tags in `csproj` PropertyGroup
```
<MagickCopyNativeLinux>true</MagickCopyNativeLinux>
<MagickCopyNativeLinuxMusl>true</MagickCopyNativeLinuxMusl>
<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles>
```

2. I am renaming `Magick.NET-Q8-x64.Native.dll.so` from `runtimes/linux-x64/native` folder to `libMagick.NET-Q8-x64.Native.dll.so` and copying to `bin\Debug\net6.0` folder.

3. Adding Below code in docker file too
```
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
        fontconfig \
        libopenjp2-7 \
        libfontconfig1 \
    && rm -rf /var/lib/apt/lists/*
```

### Docker OS information
```
Shell command output: PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=Debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
4.14.294-220.533.amzn2.x86_64 kernel version
```

### Error
```
System.TypeInitializationException: The type initializer for 'NativeMagickSettings' threw an exception.
---> System.DllNotFoundException: Unable to load shared library 'Magick.Native-Q8-x64.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libMagick.Native-Q8-x64.dll: cannot open shared object file: No such file or directory
at ImageMagick.Environment.NativeEnvironment.Initialize() in /_/src/Magick.NET/Native/Helpers/Environment.cs:line 65
at ImageMagick.Environment.Initialize() in /_/src/Magick.NET/Helpers/Environment.cs:line 21
at ImageMagick.MagickSettings.NativeMagickSettings..cctor() in /_/src/Magick.NET/Native/Settings/MagickSettings.cs:line 321
--- End of inner exception stack trace ---
at ImageMagick.MagickSettings.NativeMagickSettings..ctor() in /_/src/Magick.NET/Native/Settings/MagickSettings.cs:line 343
at ImageMagick.MagickSettings..ctor() in /_/src/Magick.NET/Settings/MagickSettings.cs:line 33
at ImageMagick.MagickImage..ctor() in /_/src/Magick.NET/MagickImage.cs:line 41
at ImageMagick.MagickImage..ctor(Byte[] data) in /_/src/Magick.NET/MagickImage.cs:line 51
````

### Code
````
using (MagickImage image = new(attachment))
{
    (int desiredWidth, int desiredHeight) = GetImageResizeSize(image.Width, image.Height);

    MagickGeometry size = new(desiredWidth, desiredHeight)
    {
        IgnoreAspectRatio = false
    };

    image.Format = MagickFormat.Jpeg;
    image.Quality = 80;

    image. Resize(size);

    MemoryStream outputImageStream = new();

    image.Write(outputImageStream);

    return outputImageStream.ToArray();
}
````

**Comment by dlemstra** [comment]
What is the output of `ldd Magick.NET-Q8-x64.Native.dll.so`?

**Comment by Manish082007** [comment]
@dlemstra - instead of renaming dll file to `libMagick.NET-Q8-x64.Native.dll.so`, I renamed it to `libMagick.NET-Q8-x64.Native.dll`. Now it able to find load the DLL.
...
But right now, I am getting error ImageCorrupt error. By looking some of issues from GitHub, i came to know there is no issue with `Magick.Net` library. I am using Aws Api Gateway with multipart content type. So i need to play with some configuration. I will keep you posted with finding.
...
Thank you

**Comment by Manish082007** [comment, marked as answer]
@dlemstra - I am able to resolve this issue. The problem was Aws Api Gateway was converting base64 string. I just need to specify the multipart-data as binary type & re-deploy the API stage. Thank you for all your help.
